scripts/ci/pre_commit/compile_www_assets.py

```diff
@@ -71,4 +72,23 @@ def get_directory_hash(directory: Path, skip_path_regexp: str | None = None) ->
     subprocess.check_call(["yarn", "install", "--frozen-lockfile"], cwd=os.fspath(www_directory))
     subprocess.check_call(["yarn", "run", "build"], cwd=os.fspath(www_directory), env=env)
     new_hash = get_directory_hash(www_directory, skip_path_regexp=r".*node_modules.*")
-    WWW_HASH_FILE.write_text(new_hash)
+    www_hash_file.write_text(new_hash)
+
+
+def is_fab_provider_installed() -> bool:
+    return (
+        importlib.util.find_spec("airflow") is not None
```

Comment:
I am not sure whether this will work. because airflow and providers are not installed in pre-commit.

But as you can be sure the source tree is checked-out, how about checking if source tree is available?

Reply:
But when you install, the sources are not available, are not they? So `is_fab_provider_installed` will return `False` in this case? Although we wan do want to generate assets for FAB as well. As for the `pre-commit`, since Airflow is not installed, `is_fab_provider_installed` will return `False`.

I am not 100% sure I understand everything correctly here so I might be wrong :)

Reply:
I thought from the point of lifecycle. The pre-commit hook will be called in CI and locally... at build time. Source tree will be there. And it _needs_ to be called at build-time to create the static JS content as dist folder to be packaged into the wheel.

The check you are referring to is at _runtime_ At `pip install apache-airflow-providers-fab` time the wheel is only un-packed to site-packages. There will no pre-commit be executed as well as you need to assume that at point of install no yarn will be available (might be the user installing is even offline w/o internet).
Otherwise - not being an expert - if the wheel install need to compile/bind/do something at point of install you need to define this in package meta data - not in a pre-commit script that is not available at install time.

Reply:
Alright. Following this approach I am at the exact same point as in #45058. I dont quite get the error. Will re-try later

Reply:
@bbovenzi or @pierrejeambrun can you help here? yarn is not really successfully compiling www assets. I assume the yarn lock file is somewhat "different" but I am not an expert why the compile fails. (Still I suspect that loading airflow a Py module is not correct because at CI/compile time the packahe is not installed (chicken+egg))

Reply:
I don't see `is_fab_provider_installed`  being called ? Where was that supposed to be ?

Reply:
I think the problem is that we need to adapt the custom asset hatchling build stage for that to also work for the plugin.

I'm not an expert there but `CustomBuild.build_standard` looks like this should be updated. For now it's only taking care of the `airflow/www` dir.

Reply:
Indeed I don't think we have the providers there, core and providers must be build separately. I don't think such routes `fab_provider_www_directory = (
        AIRFLOW_SOURCES_PATH / "providers" / "src" / "airflow" / "providers" / "fab" / "www"` exists.


Maybe we need one hook for core, one hook for the fab provider. And we need to include the provider hook into the `fab` provider build process.

Maybe @potiuk can give us some hints/ideas on what's the best way to achieve that.

Reply:
It should be embedded in the build process of provider package. This is not easy to do today because we are building providers using `flit` which is not easily extendable. But when we separate providers - to separate projects https://github.com/apache/airflow/issues/44511 - this will be much easier.